### ISSUE TITLE
LG-264: MicroProfile Health v2.2 - Implementation

### ISSUE THREAD

**Issue by donbourne** [OpenLiberty/open-liberty#10377]
Eclipse MicroProfile Health: https://github.com/eclipse/microprofile-health/releases/tag/2.2

AHA: https://bigblue.aha.io/features/LG-264
UFO link: https://ibm.box.com/s/3qpmdkqet00s2j2g7pa07w5q0g5a484r
ID: https://github.ibm.com/was-liberty/liberty-docs/issues/1783

**Comment by pgunapal** [comment]
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## List of Steps to complete or get approvals / sign-offs for Onboarding to the Liberty release (GM date)
 
Instructions:
 - Do the actions below and mark them complete in the checklist when they are done.
 - Make sure all feature readiness approvers put the appropriate tag on the epic to indicate their approval.
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE **Before Development Starts or 8 weeks before Onboarding** 
- [ ] POC Design / WAD Review Scheduled (David Chang) or N/A.
- [ ] POC Design / WAD Reviewed (Feature Owner) or N/A.
- [ ] Complete any follow-ons from the POC Review.
- [ ] Design / WAD Approval (Alasdair Nottingham) or N/A.
- [ ] No Design / No WAD Approval (Arthur De Magalhaes - cloud / Alasdair Nottingham - server) or N/A.
- [ ] SVT Requirements identified. (Epic owner / Feature owner with SVT focal point)
- [ ] ID Requirements identified. (Epic owner / Feature owner with ID focal point)
- [ ] Create a child task of the epic entitled "FAT Approval Test Summary".  Add and fill in the template as described here: https://github.ibm.com/was-liberty/WS-CD-Open/wiki/Feature-Review-(Feature-Test-Summary-Process)
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE **3 weeks before Onboarding** 
- [ ] Identify all open source libraries that are changing or are new. Work with Legal Release Services (Cass Tucker or Release PM) to get open source cleared and approved. Or N/A. (Epic Owner).   New or changed open source impacts license and Certificate of Originality.
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE ** 3 weeks before Onboarding** 
- [ ] All new or changed PII messages are checked into the integration branch, before the last translation shipment out. (Epic Owner)
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE **2 weeks before Onboarding** 
- [ ] Implementation complete. (Epic owner / Feature owner) 
- [ ] All function tests complete. Ready for FAT Approval. (Epic owner / Feature owner)
- [ ] Review all known issues for Stop Ship. (Epic owner / Feature owner / PM)
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## APPROVALS with TARGET COMPLETION DATE **2 to 1 week before Onboarding** 
Prereq: You must have the Design Approved or No Design Approved label on the GitHub Epic.
- [ ] **Accessibility** - (G Scott Johnston). Accessibility testing is complete or N/A. Approver adds label focalApproved:accessibility to the Epic in Github.
- [ ] **FAT Liberty SOE** - (Kevin Smith). SOE FATS are running successfully or N/A . Approver adds label focalApproved:fat to the Epic in Github.
- [ ] **Globalization** (Sam Wong - Liberty / Simy Cheeran - tWAS). Translation is complete or N/A. TVT - complete or N/A. Approver adds label focalApproved:globalization to the Epic in Github.
- [ ] **ID** - (Kareen Deen). Documentation work is complete or N/A . Approver adds label focalApproved:id to the Epic in Github.
- [ ] **Performance** - (Jared Anderson). Performance testing is complete with no high severity defects or N/A . Approver adds label focalApproved:performance to the Epic in Github.
- [ ] **Serviceability** - (Don Bourne). Serviceability has been addressed. 
- [ ] **STE** - (Swati Kasundra). STE chart deck is complete or N/A . Approver adds label focalApproved:ste to the Epic in Github.
- [ ] **SVT** - (Greg Ecock - Cloud, Brian Hanczaryk- APS). SVT is complete or N/A . Approver adds label focalApproved:svt to the Epic in Github.
- [ ] **Demo** - (Liberty only - Tom Evans or Chuck Bridgham). Demo is scheduled for an upcoming EOI. Approver adds label focalApproved:demo to the Epic in Github.
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## TARGET COMPLETION DATE **1 week before Onboarding** 
- [ ] No Stop Ship issues for the feature. (Epic owner / Feature owner / Release PM)
- [ ] Ship Readiness Review and Release Notes completed (Epic owner / Feature owner / Release PM)
- [ ] Github Epic and Epic's issues are closed / complete. All PRs are committed to the master branch. (Epic owner / Feature owner / Backlog Subtribe PM)
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## NOT REQUIRED FOR A FEATURE
- [ ] **OL Guides** - (Yee-Kang Chang). Assessment for OL Guides is complete or N/A. 
- [ ] **WDT** - (Leonard Theivendra). WDT work complete or N/A.
 
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## Related Deliverables TARGET COMPLETION DATE **General Availability** 
- [ ] Blog article writeup (Epic owner / Feature owner / Laura Cowen)

**Comment by pgunapal** [comment]
@gscottj There are no addition of any new or change to any existing UI. May I get your approval on Accessibility?

**Comment by pgunapal** [comment]
@jhanders34 There are no significant change in performance for this feature. The performance should be similar to mpHealth-2.0/2.1 May I get your approval on performance?

**Comment by pgunapal** [comment]
@samwatibm There are no globalization/translation required for this feature. May I get your approval for globalization?

**Comment by pgunapal** [comment]
@skasund There isn't much of a difference between mpHealth-2.0/2.1 and mpHealth-2.2, as the STE for mpHealth-2.0 covers pretty much everything for mpHealth-2.2. The minor change in mpHealth-2.2, is just minor updates to the spec API. The new spec changes will be documented in KC and OL docs. May I get the STE approval from you? Thanks!

**Comment by skasund** [comment]
@pgunapal  We don't need an STE since it'll be documented.  I've approved the feature.  Thanks!

**Comment by pgunapal** [comment]
@donbourne **Serviceability Approval Comment** - Please answer the following questions for serviceability approval:

1.  UFO -- does the UFO identify the most likely problems customers will see and identify how the feature will enable them to diagnose and solve those problems without resorting to raising a PMR?  Have these issues been addressed in the implementation?

- Yes, the UFO includes the most likely problems the customers will see, which is usually in their own implementation of the HealthCheck interface, in their applications. Detailed messages will be logged in the messages.log. No major functional change was made for this mpHealth-2.2 feature, as its implementation is similar to mpHealth-2.0 feature.

2.  Test and Demo -- As part of the serviceability process we're asking feature teams to test and analyze common problem paths for serviceability and demo those problem paths to someone not involved in the development of the feature (eg. L2, test team, or another development team).  
a) What problem paths were tested and demonstrated?
b) Who did you demo to?
c) Do the people you demo'd to agree that the serviceability of the demonstrated problem scenarios is sufficient to avoid PMRs for any problems customers are likely to encounter, or that L2 should be able to quickly address those problems without need to engage L3?

- For mpHealth-2.2, just minor spec updates were made, such as changing Abstract HealthCheckResponse class to a concrete class, in order for consumers to use. The common problem paths are similar to mpHealth-2.0, and was previously tested and demo'ed to SVT.

3.  SVT -- SVT team is often the first team to try new features and often encounters problems setting up and using them.  Note that we're not expecting SVT to do full serviceability testing -- just to sign-off on the serviceability of the problem paths they encountered.
a) Who conducted SVT tests for this feature?

- No SVT was done for this feature, as it only had minor changes compared to mpHealth-2.0/2.1. 
Brian H. from the SVT team was contacted, regarding SVT.

b) Do they agree that the serviceability of the problems they encountered is sufficient to avoid PMRs, or that L2 should be able to quickly address those problems without need to engage L3?  

- N/A

4.  Which L2 / L3 queues will handle PMRs for this feature?  Ensure they are present in the contact reference file and in the queue contact summary, and that the respective L2/L3 teams know they are supporting it.  Ask Don Bourne if you need links or more info.
- WAS L3: Liberty Log Analytics

**Comment by gscottj** [comment]
@pgunapal Focal approval granted.

**Comment by pgunapal** [comment]
@hanczaryk Since, the mpHealth-2.2 feature has only minor spec api updates, and is using the same Liberty implementation as mpHealth-2.0, is SVT still needed? If not, may I get your approval on SVT? Thanks!

**Comment by chirp1** [comment]
Prashanth opened this ID issue and added info to it: https://github.ibm.com/was-liberty/liberty-docs/issues/1783

**Comment by pgunapal** [comment]
@tevans78 Demo is scheduled for Liberty 20.5 - March 10th EOI.
